src/libos/src/util/sgx/dcap/quote_generator.rs

```diff
@@ -4,31 +4,48 @@ pub use sgx_types::{sgx_ql_qv_result_t, sgx_quote3_error_t, sgx_report_data_t, s
 pub struct QuoteGenerator {
     qe_target_info: sgx_target_info_t,
     quote_size: u32,
+    sgx_status: sgx_status_t,
+    qe3_ret: sgx_quote3_error_t,
 }
 
 impl QuoteGenerator {
     pub fn new() -> Self {
         let mut qe_target_info = sgx_target_info_t::default();
         let mut quote_size: u32 = 0;
+        let mut sgx_status = sgx_status_t::SGX_SUCCESS;
+        let mut qe3_ret = sgx_quote3_error_t::SGX_QL_SUCCESS;
 
         unsafe {
-            let mut qe3_ret = sgx_quote3_error_t::SGX_QL_SUCCESS;
-            let sgx_status = occlum_ocall_init_dcap_quote_generator(
+            sgx_status = occlum_ocall_init_dcap_quote_generator(
                 &mut qe3_ret,
                 &mut qe_target_info,
                 &mut quote_size,
             );
-            assert_eq!(sgx_status_t::SGX_SUCCESS, sgx_status);
-            assert_eq!(
-                sgx_quote3_error_t::SGX_QL_SUCCESS,
-                qe3_ret,
-                "fail to launch QE"
-            );
+
+            if sgx_status != sgx_status_t::SGX_SUCCESS {
+                error!("Init dcap quote generator failed {}", sgx_status);
+            }
+
+            if qe3_ret != sgx_quote3_error_t::SGX_QL_SUCCESS {
+                error!("DCAP fail to launch QE {}", qe3_ret);
+            }
         }
 
         Self {
             qe_target_info,
             quote_size,
+            sgx_status,
+            qe3_ret,
+        }
+    }
+
+    pub fn is_ql_healthy(&self) -> bool {
```

Comment:
Maybe the name is_healthy is better than is_ql_healthy, because it indicates the QuoteGenerator.